webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+var errMissingNewObject = errors.New("the new object may not be nil")
+
+// Callback is a generic function to be called by a consumer of validation
+type Callback struct {
+	// funcion is the callback to be invoked
+	function func(ctx context.Context, unstructured *unstructured.Unstructured) error
+
+	// supportedVerbs are the verbs supported for the callback.
+	// The function will only be called on these acitons.
+	supportedVerbs map[webhook.Operation]bool
+}
+
+// NewCallback creats a new callback function to be invoked on supported vebs.
+func NewCallback(function func(context.Context, *unstructured.Unstructured) error, supportedVerbs []webhook.Operation) (c *Callback) {
```

Comment:
```suggestion
// NewCallback creates a new callback function to be invoked on supported vebs.
func NewCallback(function func(context.Context, *unstructured.Unstructured) error, supportedVerbs []webhook.Operation) (c Callback) {
```

You can drop the pointer to simplify things